src/app/shared/components/template/components/layout/nav_group.ts

```diff
@@ -138,30 +157,63 @@ export class NavGroupComponent extends TemplateLayoutComponent {
     return result > 0 ? result : 0;
   }
 
+  /**
+   * Function that will move forward or back to Section
+   * @param index
+   */
   async goToSection(index: number) {
     this.sectionIndex = index;
     this.scrollToTop();
     this._row._debug_name = this.templateNames[index];
     await this.updateSectionProgress();
   }
 
+  /**
+   * Function to Update Progress of Stepper
+   */
   async updateSectionProgress() {
     //update the field provided in progress_variable to be equal to the max of it's current value
     //and the percentage of this.sectionIndex from this.templateNames.length. the value should
     //be an integer between 0 and 100 inclusive.
     const progressField = this._row.parameter_list["progress_field"];
+    const progressFieldMaximum = this._row.parameter_list["max_progress_field"];
+
     if (progressField && progressField.indexOf("{{") < 0) {
       const currentPercentDone = Math.ceil(
         ((this.sectionIndex + 1) / this.templateNames.length) * 100
       );
+
       const previousPercentDone = Number.parseInt(this.templateService.getField(progressField));
-      let percentDone = currentPercentDone;
-     
+      let maximumPercentDone: Number;
+      this.sectionMaxiumumPercentage = Math.max(
+        currentPercentDone,
+        this.sectionMaxiumumPercentage as number
+      );
+
+      if (previousPercentDone && !isNaN(previousPercentDone)) {
+        maximumPercentDone = Math.max(currentPercentDone, previousPercentDone);
+        if (maximumPercentDone > this.sectionMaxiumumPercentage) {
+          this.sectionMaxiumumPercentage = maximumPercentDone;
+          this.templateService.setField(progressFieldMaximum, maximumPercentDone.toString());
+        } else {
+          this.templateService.setField(
+            progressFieldMaximum,
+            this.sectionMaxiumumPercentage.toString()
+          );
+        }
+      }
```

Comment:
```suggestion
      if (previousPercentDone && !isNaN(previousPercentDone)) {
        maximumPercentDone = Math.max(currentPercentDone, previousPercentDone);
        }
      }
```
See previous comment

Reply:
Example , User reaches to Step 3 of 4.
Now currentPercentDone will be 75 and previousPercentDone will be 50, this maximumPercentDone will be 75.

User reaches to Step 2 of 4 after tapping 3 of 4.
Now currentPercentDone will be 50 and previousPercentDone will be 75, this maximumPercentDone will be 75.

User Tap Stepper Step 2 of 4 again.
Now currentPercentDone will be 50 and previousPercentDone will be 50, this maximumPercentDone will be 50 and it should be actually 75.